### DISCUSSION TITLE
There is obvious lag when the new chunk mesh is loaded. Is there any way to solve it?

### DISCUSSION THREAD

**Discussion by zzhgithub** [bevyengine/bevy#9001] [Q&A] [answered]
![image](https://github.com/bevyengine/bevy/assets/12131877/d32e0745-e59b-40f4-9ae5-db02e861036d)

here the code 
https://github.com/zzhgithub/just_join/tree/main

I used task pool. But not solve the problem.
Can anyone told me how to fix it ? Thanks

**Comment by Pieresqi** [comment]
Hi, looking at your code I think it's caused by this line:
https://github.com/zzhgithub/just_join/blob/a3e047590a3d10e7c404ebefbabbfe4721906569/src/mesh_generator.rs#L43C50-L43C50

That line causes your function to block so it stalls rest of app. You need to wrap that ele variable in poll_once function, then on it you can do block_on.

**Reply by Pieresqi** [reply, marked as answer]
So it would look like this:
`futures_lite::future::block_on(futures_lite::future::poll_once(ele))`

**Reply by zzhgithub** [reply]
Thank you
